code/game/machinery/porta_turret/portable_turret_construct.dm

```diff
@@ -40,122 +40,131 @@
 				set_anchored(TRUE)
 				build_step = PTURRET_BOLTED
 				return
 
-			else if(I.tool_behaviour == TOOL_CROWBAR && !anchored)
-				I.play_tool_sound(src, 75)
+			else if(used.tool_behaviour == TOOL_CROWBAR && !anchored)
+				used.play_tool_sound(src, 75)
 				to_chat(user, span_notice("You dismantle the turret construction."))
 				new /obj/item/stack/sheet/iron(loc, 5)
 				qdel(src)
 				return
 
 		if(PTURRET_BOLTED)
-			if(istype(I, /obj/item/stack/sheet/iron))
-				var/obj/item/stack/sheet/iron/M = I
-				if(M.use(2))
+			if(istype(used, /obj/item/stack/sheet/iron))
+				var/obj/item/stack/sheet/iron/sheet = used
+				if(sheet.use(2))
 					to_chat(user, span_notice("You add some metal armor to the interior frame."))
 					build_step = PTURRET_START_INTERNAL_ARMOUR
 					icon_state = "turret_frame2"
 				else
 					to_chat(user, span_warning("You need two sheets of iron to continue construction!"))
 				return
 
-			else if(I.tool_behaviour == TOOL_WRENCH)
-				I.play_tool_sound(src, 75)
+			else if(used.tool_behaviour == TOOL_WRENCH)
+				used.play_tool_sound(src, 75)
 				to_chat(user, span_notice("You unfasten the external bolts."))
 				set_anchored(FALSE)
 				build_step = PTURRET_UNSECURED
 				return
 
 
 		if(PTURRET_START_INTERNAL_ARMOUR)
-			if(I.tool_behaviour == TOOL_WRENCH)
-				I.play_tool_sound(src, 100)
+			if(used.tool_behaviour == TOOL_WRENCH)
+				used.play_tool_sound(src, 100)
 				to_chat(user, span_notice("You bolt the metal armor into place."))
 				build_step = PTURRET_INTERNAL_ARMOUR_ON
 				return
 
-			else if(I.tool_behaviour == TOOL_WELDER)
-				if(!I.tool_start_check(user, amount = 5)) //uses up 5 fuel
+			else if(used.tool_behaviour == TOOL_WELDER)
+				if(!used.tool_start_check(user, amount = 5)) //uses up 5 fuel
 					return
 
 				to_chat(user, span_notice("You start to remove the turret's interior metal armor..."))
 
-				if(I.use_tool(src, user, 20, volume = 50, amount = 5)) //uses up 5 fuel
+				if(used.use_tool(src, user, 20, volume = 50, amount = 5)) //uses up 5 fuel
 					build_step = PTURRET_BOLTED
 					to_chat(user, span_notice("You remove the turret's interior metal armor."))
 					new /obj/item/stack/sheet/iron(drop_location(), 2)
 					return
 
 
 		if(PTURRET_INTERNAL_ARMOUR_ON)
+<<<<<<< HEAD
 			if(istype(I, /obj/item/gun/energy)) //the gun installation part
 				if(istype(I, /obj/item/gun/energy/cell_loaded/medigun)) //SKYRAT EDIT MEDIGUNS//
 					to_chat(user, span_notice("The [I] is unable to fit inside of the turret"))
 					return //SKYRAT EDIT MEDIGUN END//
 				var/obj/item/gun/energy/E = I
 				if(!user.transferItemToLoc(E, src))
+=======
+			if(istype(used, /obj/item/gun/energy)) //the gun installation part
+				var/obj/item/gun/energy/egun = used
+				if(egun.gun_flags & TURRET_INCOMPATIBLE)
+					to_chat(user, span_notice("You don't think it would be right to add [used] to the turret"))
+>>>>>>> 642722affa4 (Adds a blacklist for putting guns in turrets (#75263))
```

Comment:
```suggestion
			if(istype(used, /obj/item/gun/energy)) //the gun installation part
				var/obj/item/gun/energy/egun = used
				if(egun.gun_flags & TURRET_INCOMPATIBLE)
					to_chat(user, span_notice("You don't think it would be right to add [used] to the turret"))
```